sources/niLang/src/System_Linux.cpp

```diff
@@ -193,8 +205,18 @@ struct sX11System : public Impl_HeapAlloc {
     sRecti mrectMonitor;
     tOSMonitorFlags mFlags;
   };
+
+  struct sX11Clipboard {
+    Ptr<iCallback> mCallback;
+    Ptr<iDataTable> mData;
+    tBool mbWaiting;
+  } mClipboard;
+
   astl::vector<sX11Monitor> mvMonitors;
   tF32 mfContentsScale = 1.0f;
+  // wee need window and display for clipboard synchronization
+
+  Ptr<cLinuxWindow> mpWindow;
```

Comment:
`mp` is for `raw pointers - cLinuxWindow*`, it should be `mptr for Ptr<T>`. But ultimately this should be a weak pointer not an owning pointer so `WeakPtr<cLinuxWindows> mwWindow`, use `QPtr<>` when you need the instance to check that its still there when you need it.